File: SimpleCommerce.Domain/Entities/Address.cs
namespace SimpleCommerce.Domain.Entities;

public class Address
{
    public string Id { get; set; } = string.Empty;
    public string CustomerId { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string FullName { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string District { get; set; } = string.Empty;
    public string AddressLine { get; set; } = string.Empty;
    public string? ZipCode { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: simplecommerce-frontend/src/types/address.ts
export interface Address {
  id: string
  title: string
  fullName: string
  phone: string
  city: string
  district: string
  addressLine: string
  zipCode?: string | null
}

export interface CreateAddressRequest {
  customerId: string
  title: string
  fullName: string
  phone: string
  city: string
  district: string
  addressLine: string
  zipCode?: string | null
}

export interface UpdateAddressRequest {
  title: string
  fullName: string
  phone: string
  city: string
  district: string
  addressLine: string
  zipCode?: string | null
}
